src/ploneintranet/workspace/tests/test_views.py

```diff
@@ -154,3 +157,37 @@ def test_acquired_roles_from_policy_settings(self):
         self.assertTrue(
             all([results[0]["roles"][role] == "acquired" for role in roles]),
             "Acquired roles were not set correctly")
+
+
+class TestFileUploadView(BaseViewTest):
+    def setUp(self):
+        super(TestFileUploadView, self).setUp()
+        self.view = FileUploadView(self.workspace, self.request)
+
+    def test_no_file_json(self):
+        self.request.environ['HTTP_ACCEPT'] = 'text/json'
+        self.assertEqual(self.view(), {})
+
+    def test_no_file_html(self):
+        self.request.environ['HTTP_ACCEPT'] = 'text/html'
+        self.assertEqual(self.view(), None)
+        self.assertEqual(self.request.response.getStatus(), 302)
+
+    def test_upload_file_json(self):
+        self.request.environ['HTTP_ACCEPT'] = 'text/json'
+        self.request.form = {'file': 'dummy'}
+        from ploneintranet.workspace.browser.views import BaseFileUploadView
```

Comment:
I'm not sure I understand why there are repeated imports within these methods. I understand the reasoning for putting imports into a test method, but as we haven't been following that pattern, it seems odd to start now.